core/db/entities.py

```diff
@@ -289,16 +323,45 @@ def itl_regions(self):
         return distinct_regions
 
     @classmethod
-    def get_project_by_programme_ids_and_submission_ids(
-        cls, programme_ids: list, submission_ids: list
-    ) -> list["Project"]:
-        """Get projects based on a list of programme IDs and submission IDs.
+    def get_projects_by_programme_ids_and_submission_ids(cls, programme_ids, submission_ids):
+        """Get projects that match both programme id and submission ids.
 
         :param programme_ids: A list of programme IDs to filter projects.
         :param submission_ids: A list of submission IDs to filter projects.
-        :return: A list of projects matching the provided programme IDs and submission IDs.
+        :return:
+        """
+        projects = cls.query.filter(
+            and_(cls.programme_id.in_(programme_ids), cls.submission_id.in_(submission_ids))
+        ).all()
+        return projects
+
+    @classmethod
+    def filter_projects_by_outcome_categories(
+        cls, projects: list["Project"], outcome_categories: list
+    ) -> tuple[list["Project"], list["OutcomeData"]]:
+        """Filter projects by outcome categories.
```

Comment:
Could you just elaborate a little on the docstring as to what the method is doing, ta

Reply:
Added